app/src/components/ContractOwnerCard.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+  <CardComponent title="">
+    <div v-if="ownerUser.address" class="flex justify-between items-center">
+      <div class="text-lg text-gray-500">
+        <span class="flex items-center gap-4">
+          <UserAvatarComponent :user="ownerUser" class="" />
+        </span>
+      </div>
+
+      <div class="text-lg text-gray-500">
+        <span class="flex items-center gap-4">
+          Owner address:
+          <AddressToolTip :address="ownerUser.address" />
+        </span>
+      </div>
+    </div>
+
+    <div v-else class="flex justify-center items-center py-4">
+      <span class="text-gray-500">Loading owner information...</span>
+    </div>
+  </CardComponent>
+</template>
+
+<script setup lang="ts">
+import UserAvatarComponent from '@/components/UserAvatarComponent.vue'
+import AddressToolTip from '@/components/AddressToolTip.vue'
+import { useTeamStore } from '@/stores/'
+import { type User } from '@/types/user'
+import type { Member } from '@/types/member'
+import { ref, watch, computed } from 'vue'
+import { readContract } from '@wagmi/core'
+import { config } from '@/wagmi.config'
+import type { Abi, Address } from 'viem'
+import CardComponent from '@/components/CardComponent.vue'
+
+const props = defineProps<{
+  contractAddress?: Address
+}>()
+
+const teamStore = useTeamStore()
+
+// Minimal ABI to read owner() from arbitrary contract
+const OWNER_ABI: Abi = [
+  {
+    name: 'owner',
+    stateMutability: 'view',
+    type: 'function',
+    inputs: [],
+    outputs: [{ type: 'address' }]
+  }
+]
+
+const ownerUser = ref<User>({
+  name: '',
+  address: undefined
+})
+
+// prefer explicit prop
+const contractAddress = computed<Address | undefined>(
+  () => props.contractAddress as Address | undefined
+)
+
+const resolveOwnerFromContract = async () => {
+  const contractAddr = contractAddress.value!
+
+  try {
+    const chainOwner = await readContract(config, {
+      address: contractAddr as `0x${string}`,
```

Comment:
If contract Address is an Address type, you don't need again as `0x${string}`,
```suggestion
      address: contractAddr,
```